python/lsst/pipe/analysis/visitAnalysis.py

```diff
@@ -396,6 +457,16 @@ def readCatalogs(self, dataRefList, dataset, repoInfo, aliasDictList=None):
                                                  "Id of CCD on which source was detected")
 
             # Compute Focal Plane coordinates for each source if not already there
+            if self.config.hasFakes:
+                exp = repoInfo.butler.get("calexp", dataRef.dataId)
+                mask = exp.mask.array
+                maskBad = mask & 2**0
+                bad = len(np.where((maskBad == 1))[0])
+                good = len(np.where((maskBad == 0))[0])
+                pixScale = exp.getWcs().getPixelScale().asArcseconds()
+                area = pixScale**2 * good
+                areaDict[dataRef.dataId["ccd"]] = area
```

Comment:
I see you have added this `areaDict` you are passing around, but I'm not sure what it's for? Keeping track of how much area on the image fake sources use up? If areaDict could just be None for situations without fakes, that might simplify the new if/else statements you had to introduce.

Reply:
areaDict contains that usable area of each ccd derived from the mask plane. I don't really want to change the outputs of readCatalogs in general because I am not sure where else it is used and if it will affect anything.